File: JavaStuff/Go/SpieglSpieglAnDerWand.go
package main

import (
	"fmt"
)

func reverse( str string ) string {
	var result string
    for _, v := range str { 
        result = string(v) + result 
	} 
	return result
}

func main() {
	var str string
	for ( true ) {
		fmt.Scan( &str )
		if ( str == "quit" ) {
			break;
		}
		if (str == "" ) {
			str = "Spieglein Spieglein an der Wand";
		}
		str = reverse(str)
		fmt.Println(str)
	}
}

File: JavaStuff/SpieglSpieglAnDerWand.java
import java.util.Scanner;


public class SpieglSpieglAnDerWand {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String str;
        StringBuffer strb;

        while (true)
        {
            str = scan.nextLine();
            if ( str.equals("quit()") ) 
            {
                break;
            }
            if (str.equals("") )
            {
                str = "Spieglein Spieglein an der Wand";
            }
            strb = new StringBuffer(str);
            System.out.println(strb.reverse());
        }
        scan.close();
    }
}
